crates/bevy_utils/src/lib.rs

```diff
@@ -45,21 +45,33 @@ use hashbrown::hash_map::RawEntryMut;
 use std::{
     any::TypeId,
     fmt::Debug,
-    future::Future,
     hash::{BuildHasher, BuildHasherDefault, Hash, Hasher},
     marker::PhantomData,
     mem::ManuallyDrop,
     ops::Deref,
-    pin::Pin,
 };
 
-/// An owned and dynamically typed Future used when you can't statically type your result or need to add some indirection.
 #[cfg(not(target_arch = "wasm32"))]
-pub type BoxedFuture<'a, T> = Pin<Box<dyn Future<Output = T> + Send + 'a>>;
+mod conditional_send {
+    /// Use [`ConditionalSend`] to mark an optional Send trait bound as optional, as on certain platforms (eg. WASM),
+    /// futures aren't Send.
+    pub trait ConditionalSend: Send {}
+    impl<T: Send> ConditionalSend for T {}
+
+    /// An owned and dynamically typed Future used when you can't statically type your result or need to add some indirection.
+    pub type BoxedFuture<'a, T> =
+        std::pin::Pin<Box<dyn std::future::Future<Output = T> + Send + 'a>>;
```

Comment:
Could both these definitions be merged into a single definition using `ConditionalSend` instead of `Send`?

```rust
pub type BoxedFuture<'a, T> = std::pin::Pin<Box<dyn std::future::Future<Output = T> + ConditionalSend + 'a>>;
```

Reply:
Thanks! Alas can't quite do that directly as `only auto traits can be used as additional traits in a trait object` :/ [Trait aliases ](https://rust-lang.github.io/rfcs/1733-trait-alias.html#:~:text=Conceptually%2C%20a%20trait%20alias%20is,parameters%20list%2C%20where%2Dclause))would make this easy.

There's another way to do it though with a new trait + blanket impl, gave that a shot, does at least pull things out of the conditional compiled block!